syntax = "proto3";

package saq.sandwich.proto.api.v1;


import "proto/api/v1/encoding_format.proto";

// Data source consisting of either a file or an inline value.
message DataSource {
  oneof specifier {
    // Local filesystem data source.
    string filename = 1;

    // Bytes inlined in the configuration.
    bytes inline_bytes = 2;

    // String inlined in the configuration.
    string inline_string = 3;
  }
}

// An ASN1 encoded data source
message ASN1DataSource {
  // The raw data source
  DataSource data = 1;

  // The ASN1 encoding format of that data source
  ASN1EncodingFormat format = 2;
}
